src/main/java/gui/viewmodel/MainPageViewModel.java

```diff
@@ -0,0 +1,163 @@
+package gui.viewmodel;
+
+
+import datagateway.event.EventReader;
+import datagateway.event.ObservableEventRepository;
+import datagateway.task.ObservableTaskRepository;
+import datagateway.task.TaskReader;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+
+public class MainPageViewModel extends ViewModel implements Runnable{
+    private final ObservableTaskRepository taskRepository;
+    private final ObservableEventRepository eventRepository;
+    private final List<TaskReader> relevantTasks = new ArrayList<>();
+    private final List<EventReader> relevantEvents = new ArrayList<>();
+    private final List<Consumer<Readers>> observers = new ArrayList<>();
+
+    public MainPageViewModel(ObservableTaskRepository observableTaskRepository, ObservableEventRepository observableEventRepository){
+        this.taskRepository = observableTaskRepository;
+        this.eventRepository = observableEventRepository;
+
+        observableTaskRepository.addCreationObserver(this::handleCreation);
```

Comment:
repository binding was implemented during construction in the #149 PR. No need to do this anymore. Instead get the `TaskRequestBoundary` or whatever the return type of `serviceFactory.makeTaskGetter()` is, and use that instead of the repository.